src/rgw/rgw_rest_pubsub.cc

```diff
@@ -422,6 +422,11 @@ void RGWPSCreateTopicOp::execute(optional_yield y) {
                          << op_ret << dendl;
       return;
     }
+  } else if (already_persistent) {  // redundant call to CreateTopic
+    // initialize the persistent queue's location, using ':' as the namespace
+    // delimiter because its inclusion in a TopicName would break ARNs
+    dest.persistent_queue =
+        string_cat_reserve(get_account_or_tenant(s->owner.id), ":", topic_name);
```

Comment:
i think this should just copy `dest.persistent_queue = topic->dest.persistent_queue`. the topic we're overwriting may have been a legacy one whose queue doesn't include the tenant namespace

Reply:
would it not be better in way, the user now gets migrated to correct queue name ?
just that old queue will remain there in the system

Reply:
not unless we have a way to clean up the old queue. the notification thread would keep polling on it forever

Reply:
may be this code will still not make use of the new queue name even if it exist 
```    
if (struct_v >= 7) {
      decode(persistent_queue, bl);
    } else if (persistent) {
      // persistent topics created before v7 did not support tenant namespacing.
      // continue to use 'arn_topic' alone as the queue's rados object name
      persistent_queue = arn_topic;
    }

Reply:
> i think this should just copy `dest.persistent_queue = topic->dest.persistent_queue`. the topic we're overwriting may have been a legacy one whose queue doesn't include the tenant namespace

done